playkit/src/main/java/com/kaltura/playkit/LocalAssetsManager.java

```diff
@@ -233,9 +256,64 @@ public void run() {
         });
     }
 
+    private void checkClearAssetStatus(String localAssetPath, String assetId, AssetStatusListener listener) {
+        try {
+            localDataStore.load(buildAssetKey(assetId));
+            listener.onStatus(localAssetPath, 0, 0, true);
+        } catch (FileNotFoundException e) {
+            listener.onStatus(localAssetPath, 0, 0, false);
+        }
+    }
+
+    private static PKDrmParams findSupportedDrmParams(@NonNull PKMediaSource mediaSource) {
+        if (mediaSource.getDrmData() == null) {
+            return null;
+        }
+
+        for (PKDrmParams params : mediaSource.getDrmData()) {
+            switch (params.getScheme()) {
+                case WidevineCENC:
+                    if (MediaSupport.widevineModular()) {
+                        return params;
+                    }
+                    break;
+                case WidevineClassic:
+                    if (MediaSupport.widevineClassic()) {
+                        return params;
+                    }
+                    break;
+                case PlayReadyCENC:
+                    log.d("Skipping unsupported PlayReady params");
+                    break;
+            }
+        }
+        return null;
+    }
+
+    @Nullable
+    private PKDrmParams.Scheme getLocalAssetScheme(@NonNull String assetId) {
+        PKMediaFormat format;
+
+        try {
+            String formatName = new String(localDataStore.load(buildAssetKey(assetId)));
+            format = PKMediaFormat.valueOf(formatName);
+        } catch (FileNotFoundException e) {
+            log.e(e.getMessage());
+            return null;
+        }
+
+        switch (format) {
+            case dash:
+                return PKDrmParams.Scheme.WidevineCENC;
```

Comment:
Dash can also be PlayReady scheme, you must store the scheme too. I suggest that the storage entry contains "format:scheme", for example "dash:WidevineCENC".